errorset/errorset.go

```diff
@@ -20,7 +20,7 @@ import (
 )
 
 func Append(err error, errs ...error) error {
-	if err == nil && len(errs) == 1 && errs[0] == nil {
+	if err == nil && (len(errs) == 0 || len(errs) == 1 && errs[0] == nil) {
```

Comment:
I don't fully understand this and the corresponding test. So technically you could call Append(nil, nil, nil) and it should still return nil but that's not checked here.

Reply:
Could this be changed to call Append() recursively to make it more generic?

Reply:
Yeah, I know. I was being lazy. I'll do this proper.

Reply:
Thanks! ;-)